model/validation.go

```diff
@@ -669,3 +661,89 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 	return errs
 }
+
+func validateProxyAddress(hostAddr string) error {
```

Comment:
I don't think we should be resolving DNS names in validation. First, DNS resolvers might be different between golang and envoy. Second, DNS can change the IP assignment.